crates/next-core/js/src/entry/page-loader.ts

```diff
@@ -0,0 +1,16 @@
+import * as page from ".";
+
+// inserted by rust code
+declare const PAGE_PATH: string;
+
+(window.__NEXT_P = window.__NEXT_P || []).push([
+  PAGE_PATH,
+  function () {
+    return page;
+  },
+]);
+if (module.hot) {
+  module.hot.dispose(function () {
+    window.__NEXT_P.push([PAGE_PATH]);
```

Comment:
Please add a comment explaining why this is necessary for HMR.

Reply:
I'm not entirely sure, copied it from next.js 🙃